### PR TITLE
tool "Assets/web3/coin/presale/admin"

### PR DESCRIPTION
**Pull request by artman325**
- change format in `platform/plugins/Assets/config/plugin.json`, section: Users.web3.contracts
- created `Q.Users.Web3.validate` sections for validate user input in tools
- created tool "Assets/web3/coin/presale/admin"
With this tool, anyone can produce [FundContract](https://github.com/Intercoin/fundContract/) and already have a view created.

Example to initiate
```php 
echo Q::tool("Assets/web3/coin/presale/admin", array(
        "chainId" => "0x13881"
)) 
```
 
 P.S. such pull request contain changes on master branch, see [merge with master branch](https://github.com/Qbix/Platform/commit/661b87b514cea1c439db95b38a07355879e927d5)

### CODE REVIEW

File: platform/plugins/Assets/text/Assets/content/en.json [hunk 1]
```diff
@@ -626,6 +626,83 @@
...
+						"timestamps": "Timestamps",
+						"prices": "Prices",
+						"endTime": "End Time",
+						"thresholds": "Thresholds",
+						"bonuses": "Bonuses",
+						"ownerCanWithdraw": "Withdraw possibility",
+						"whitelistData": "whitelist Data",
+						"communityAddress": "Community address",
+						"methodkeccack": "Keccak256 hash function",
+						"roleId": "Role id"
```
Comment (EGreg) [inline platform/plugins/Assets/text/Assets/content/en.json:643]: Role ID

File: platform/plugins/Assets/text/Assets/content/en.json [hunk 2]
```diff
@@ -626,6 +626,83 @@
...
+					},
+					"small": {
+						"sellingToken": "Selling Token",
+						"timestamps": "Untiltime",
```
Comment (EGreg) [inline platform/plugins/Assets/text/Assets/content/en.json:647]: Did you mean to say "Until Time"

File: platform/plugins/Assets/text/Assets/content/en.json [hunk 3]
```diff
@@ -626,6 +626,83 @@
...
+						"prices": "ETH, Price",
+						"endTime": "End Time",
+						"thresholds": "token, Threshold",
```
Comment (EGreg) [inline platform/plugins/Assets/text/Assets/content/en.json:650]: "Token, Threshold" ?

File: platform/plugins/Assets/text/Assets/content/en.json [hunk 4]
```diff
@@ -626,6 +626,83 @@
...
+						"bonuses": "% Bonus",
+						"ownerCanWithdraw": "Owner Can Withdraw",
+						"whitelistData": "whitelist Data",
```
Comment (EGreg) [inline platform/plugins/Assets/text/Assets/content/en.json:653]: Did you mean "Whitelist Data"?

What is this, some geeky info ? It needs to be user friendly.

File: platform/plugins/Assets/text/Assets/content/en.json [hunk 5]
```diff
@@ -626,6 +626,83 @@
...
+						"communityAddress": "Community address",
+						"methodkeccack": "4bytes hash of method to check",
```
Comment (EGreg) [inline platform/plugins/Assets/text/Assets/content/en.json:655]: why would this be a user-facing string? can you remove all this stuff if it's not in the final UX?

File: platform/plugins/Assets/text/Assets/content/en.json [hunk 6]
```diff
@@ -626,6 +626,83 @@
...
+						"roleId": "index of role id"
+					},
+					"fields": {
+						"ownerWithdrawOptions": {
+							"option0":"Owner can not withdraw",
```
Comment (EGreg) [inline platform/plugins/Assets/text/Assets/content/en.json:660]: Owner can't withdraw (everywhere you say can not)

File: platform/plugins/Assets/web/js/Assets.js [hunk 7]
```diff
@@ -1773,6 +1838,18 @@
...
 		},
+		"Assets/web3/coin/presale/admin": {
+			js: [
+				"{{Assets}}/js/tools/web3/coin/presale/admin.js",
+				'{{Q}}/pickadate/picker.js',
+				'{{Q}}/pickadate/picker.date.js'
+			],
+			css: [
+				"{{Assets}}/css/tools/web3/coin/presale/admin.css",
+				'{{Q}}/pickadate/themes/default.css',
```
Comment (EGreg) [inline platform/plugins/Assets/web/js/Assets.js:1849]: Are you still using pickadate?

**Review by EGreg — CHANGES_REQUESTED** [review]
Please see comments, they mostly have to do with cleaning up the text files

**Comment by artman325** [comment]
it's old version tool "Assets/web3/coin/presale/admin".
after all fixes and redesign will be in pull request [tools: "Assets/web3/coin/presale/admin" and "Assets/web3/coin/presale/buy"](https://github.com/Qbix/Platform/pull/2110)

**Pull request closed without merging**